sovryn-token-bridge/federator/src/lib/Federator.js

```diff
@@ -415,4 +456,35 @@ module.exports = class Federator {
     async _getCurrentBlockNumber() {
         return this.mainWeb3.eth.getBlockNumber();
     }
+
+    async signTransaction({ blockNumber, id }) {
```

Comment:
This doesn't check block confirmations with `_isConfirmed` so the leader can get a signature even for logs that don't have enough confirmations according to the config.